packages/block-library/src/page-list/edit.js

```diff
@@ -42,7 +42,7 @@ import { useToolsPanelDropdownMenuProps } from '../utils/hooks';
 
 // We only show the edit option when page count is <= MAX_PAGE_COUNT
 // Performance of Navigation Links is not good past this value.
-const MAX_PAGE_COUNT = 100;
+const MAX_PAGE_COUNT = 500;
```

Comment:
The `100` is the default limit of the REST API; changing it to 500 would result in an error. Additionally, we can't use unbound query here due to the performance issue noted above